cli/src/ui.rs

```diff
@@ -329,10 +330,10 @@ impl PagerConfig {
             return Ok(Self::Disabled);
         };
         let args: CommandNameAndArgs = config.get("ui.pager")?;
-        if args.as_str() == Some(BUILTIN_PAGER_NAME) {
-            Ok(Self::Builtin(config.get("ui.streampager")?))
-        } else {
-            Ok(Self::External(args))
+        match (args.as_str(), env::var("PAGER")) {
+            (Some(BUILTIN_PAGER_NAME), _) => Ok(Self::Builtin(config.get("ui.streampager")?)),
+            (Some(ENV_PAGER_NAME), Ok(pager_env)) => Ok(Self::External((&pager_env).into())),
```

Comment:
It's probably better to error if `env::var("PAGER")` is not OK. This might require finagling with the error types a little bit, not sure if you tried and decided it's infeasible.